File: algo/src/main/java/com/hartwig/actin/algo/doid/datamodel/DoidEntry.java
package com.hartwig.actin.algo.doid.datamodel;

import java.util.List;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(allParameters = true,
             passAnnotations = { NotNull.class, Nullable.class })
public abstract class DoidEntry {

    @NotNull
    public abstract String id();

    @NotNull
    public abstract List<Node> nodes();

    @NotNull
    public abstract List<Edge> edges();

    @NotNull
    public abstract GraphMetadata metadata();

    @Nullable
    public abstract List<LogicalDefinitionAxioms> logicalDefinitionAxioms();

    @Nullable
    public abstract List<String> equivalentNodesSets();

    @Nullable
    public abstract List<String> domainRangeAxioms();

    @Nullable
    public abstract List<String> propertyChainAxioms();

}

File: common/src/main/kotlin/com/hartwig/actin/doid/datamodel/DoidEntry.kt
package com.hartwig.actin.doid.datamodel

data class DoidEntry(
    val id: String,
    val nodes: List<Node>,
    val edges: List<Edge>,
    val metadata: GraphMetadata,
    val logicalDefinitionAxioms: List<LogicalDefinitionAxioms>? = null,
    val equivalentNodesSets: List<String>? = null,
    val domainRangeAxioms: List<String>? = null,
    val propertyChainAxioms: List<String>? = null
)
